even_odd_lists.cpp

```diff
@@ -0,0 +1,38 @@
+/* This C++ program gets a length and a list of numbers
+from the user and prints the odd and even numbers separately.
+*/
+
+#include <iostream>
+
+using namespace std;
+
+int main() {
+    int length;
+    cout << "How many numbers? ";
+    cin >> length;
+
+    int numbers[length];

```

Comment:
strange! this should cause a compile-time error. I thought the length of an array (the number between brackets) has to be a constant of some sort. I'm sure this was the way it was explained in textbooks :D Could you please do a bit of research into this?